server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/SearchDataLoader.java

```diff
@@ -221,30 +234,15 @@ private static boolean hasLanguage(SearchWsRequest request) {
     return request.getLanguage() != null;
   }
 
-  private enum QProfileComparator implements Comparator<QProfile> {
-    INSTANCE;
-    @Override
-    public int compare(QProfile o1, QProfile o2) {
-      return new CompareToBuilder()
-        .append(o1.language(), o2.language())
-        .append(o1.name(), o2.name())
-        .toComparison();
-    }
-  }
-
-  private enum QualityProfileDtoToQProfile implements Function<QualityProfileDto, QProfile> {
-    INSTANCE;
-
-    @Override
-    public QProfile apply(QualityProfileDto input) {
-      return new QProfile()
-        .setKey(input.getKey())
-        .setName(input.getName())
-        .setLanguage(input.getLanguage())
-        .setDefault(input.isDefault())
-        .setRulesUpdatedAt(input.getRulesUpdatedAt())
-        .setLastUsed(input.getLastUsed())
-        .setUserUpdatedAt(input.getUserUpdatedAt());
-    }
+  public static QProfile qualityProfileDtoToQProfile(QualityProfileDto input, OrganizationDto organization) {
```

Comment:
how does this method compare to `QProfile.from`?
seems to me this one just doesn't copy the parent and id fields, which does look like a bug